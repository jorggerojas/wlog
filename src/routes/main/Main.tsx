import React, { useEffect, useState } from "react";
import ReactPaginate from "react-paginate";
import axios from "axios";
import t from "typy";
import PostMinified from "../post/components/PostMinified";
import { parseDate, URL } from "../../config";
import Loading from "../loading/Loading";
import { LoadingContainer } from "../../styles/containers";

const Main = () => {
  const [posts, setPosts] = useState([]);
  const [loading, setLoading] = useState(false);
  const [pageCountPosts, setpageCountPost] = useState(1);
  const getPosts = (page?: any) => {
    return axios.get(`${URL}/users/all/posts?size=6&page=${page ?? 0}`);
  };
  const changeContent = ({ selected }: any) => {
    getPosts(selected)
      .then(({ data }: any) => {
        setPosts(data.content);
      })
      .catch(() => {
        setPosts([]);
      });
  };
  useEffect(() => {
    setLoading(true);
    getPosts()
      .then(({ data }: any) => {
        setPosts(data.content);
        setpageCountPost(Math.ceil(data.totalElements / 6));
        setTimeout(() => {
          setLoading(false);
        }, 200);
      })
      .catch(() => {
        setPosts([]);
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  if (t(loading).isTrue) {
    return (
      <LoadingContainer>
        <Loading load={loading} />
      </LoadingContainer>
    );
  }
  return (
    <div className={`${t(loading).isTrue ? "backLoad" : "transparent"}`}>
      <div
        className="uk-grid uk-child-width-1-1@s uk-child-width-1-2@m uk-child-width-1-3@l uk-child-width-1-4@xl uk-padding-large"
        uk-grid="masonry: true"
        uk-height-match="target: .card-target"
      >
        {!t(posts).isEmptyArray && posts.length ? (
          posts.map((post: any) => (
            <PostMinified
              key={post.index}
              large={true}
              title={post.title}
              date={parseDate(post.dateLog)}
              user={post.user}
              summary={post.summary}
              badge={post.keywords[0]}
              id={post.index}
            />
          ))
        ) : (
          <div className="uk-text-center uk-width-1-1">
            <p className="uk-text-center uk-text-italic">
              No hay post para mostrar
            </p>
          </div>
        )}
      </div>
      <div className="paginate uk-padding">
        <ReactPaginate
          pageCount={pageCountPosts}
          onPageChange={changeContent}
          pageRangeDisplayed={2}
          marginPagesDisplayed={4}
          breakLabel="..."
          activeClassName="uk-active active"
          activeLinkClassName="uk-active active"
          disabledClassName="uk-disabled"
          nextClassName="uk-pagination-next"
          previousClassName="uk-pagination-previous"
          pageClassName=""
          containerClassName="uk-pagination uk-flex-center"
          nextLabel="Next"
        />
      </div>
    </div>
  );
};

export default Main;
