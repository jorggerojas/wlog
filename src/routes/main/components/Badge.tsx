// @flow
import React from "react";
import { Link, BrowserRouter as Router } from "react-router-dom";

interface BadgeProps {
  title: string;
  isActive: boolean;
  link?: boolean;
  target?: string;
}
const Badge = ({ title, isActive, link, target }: BadgeProps) => {
  return (
    <li key={title} className={isActive ? "uk-active" : ""}>
      <Router>
        {link ? (
          <Link to="" uk-switcher-item={target ? `${target}` : ""}>
            {title}
          </Link>
        ) : (
          <Link to={""}>{title}</Link>
        )}
      </Router>
    </li>
  );
};

export default Badge;
