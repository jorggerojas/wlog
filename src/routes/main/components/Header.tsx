import React, { useState } from "react";
import { /*globalDark,*/ globalLight } from "../../../styles/global";
import { Link } from "react-router-dom";
import cookie from "react-cookies";

const Header = () => {
  const [user] = useState(cookie.load("USER", false) ?? "0");
  const { navColor, padd, whiteText } = globalLight();
  return (
    <div uk-sticky="sel-target: .uk-navbar-container; cls-active: uk-navbar-sticky">
      <nav
        className={`${navColor} uk-navbar uk-navbar-container uk-margin`}
        uk-navbar="mode:click"
      >
        <div className="uk-navbar-left">
          <Link className={`${whiteText} uk-navbar-item uk-logo`} to="/">
            Wlog
          </Link>
        </div>
        <div className="uk-navbar-right">
          <ul className="uk-navbar-nav">
            <li className={padd}>
              {user === "0" ? (
                <Link to="/sign" className={whiteText}>
                  Iniciar sesi√≥n
                </Link>
              ) : (
                <Link to={`user/${user}`} className={whiteText}>
                  {user.toUpperCase()}
                </Link>
              )}
            </li>
          </ul>
        </div>
        <div className="uk-navbar-center">
          <Link className={`uk-navbar-toggle ${whiteText}`} to="#">
            <span uk-icon="icon:plus-circle"></span>
          </Link>
        </div>
      </nav>
    </div>
  );
};

export default Header;
