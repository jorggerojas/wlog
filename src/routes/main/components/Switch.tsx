import React from "react";

interface SwitchProps {
  check: boolean;
  checked: Function;
}

const Switch = ({ check, checked }: SwitchProps) => {
  const change = () => {
    checked(!check);
  };
  return (
    <div className="mid">
      <label className="rocker rocker-small">
        <input type="checkbox" checked={check} onChange={change} />
        <span className="switch-left">🌚</span>
        <span className="switch-right">🌞</span>
      </label>
    </div>
  );
};

export default Switch;
