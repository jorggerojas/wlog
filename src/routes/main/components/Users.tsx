import axios from "axios";
import React, { useEffect, useState } from "react";
import DataTable from "react-data-table-component";
import { loadStorage, URL } from "../../../config";
import { options, customStyles } from "./tableOptions";
import { Select, Option } from "../../../styles/text";

interface UserProps {
  theme?: boolean;
}

const Users = ({ theme }: UserProps) => {
  const [disabled, setDisabled] = useState(false);
  const [data, setData] = useState([]);
  const [totalElements, setTotalElements] = useState(0);
  const getUsers = (page?: number) => {
    return axios.get(`${URL}/users?page=${page ?? 0}&size=10`);
  };
  const changeValue = (value: string, username: string) => {
    setDisabled(true);
    setTimeout(() => {
      axios({
        method: "put",
        url: `${URL}/users/${username}`,
        headers: {
          Authorization: `Bearer ${loadStorage("TOKEN")}`,
        },
        data: {
          role: [`${value}`],
        },
      }).then(() => {
        getUsers()
          .then(({ data }: any) => {
            setData(
              data.content.map((user: any) => {
                return {
                  key: user.nickname,
                  role: user.role[0],
                  name: user.name,
                  username: user.nickname,
                  datelog: user.dateLog,
                };
              })
            );
            setTotalElements(data.totalElements);
            setDisabled(false);
          })
          .catch(() => {
            setData([]);
          });
      });
    }, 500);
  };
  const columns = [
    {
      name: "Role",
      selector: "role",
      sortable: true,
      cell: (row: any) => {
        return (
          <Select
            defaultValue={row.role ?? "REDACTOR"}
            className={" uk-text-bold"}
            onChange={(e) => changeValue(e.target.value, row.username ?? "...")}
          >
            <Option value="ADMIN">ADMIN</Option>
            <Option value="REDACTOR">REDACTOR</Option>
            <Option value="MODERADOR">MODERADOR</Option>
            <Option value="LECTOR">LECTOR</Option>
          </Select>
        );
      },
    },
    {
      name: "Full name",
      selector: "name",
      hide: 599,
      sortable: true,
      cell: (row: any) => {
        return row ? (
          <div className="uk-flex uk-flex-middle uk-flex-center">
            <p className="uk-text-bold">{row.name}</p>
          </div>
        ) : null;
      },
    },
    {
      name: "Username",
      selector: "username",
      sortable: true,
      cell: (row: any) => {
        return row ? (
          <p className="uk-text-bold">
            {row && row.username ? row.username.toUpperCase() : "..."}
          </p>
        ) : null;
      },
    },
    {
      name: "Date log",
      selector: "datelog",
      sortable: true,
      hide: 959,
      cell: (row: any) => {
        return row ? (
          <p className="uk-text-bold">{row ? row.datelog : "..."}</p>
        ) : null;
      },
    },
  ];
  useEffect(() => {
    getUsers()
      .then(({ data }: any) => {
        setData(
          data.content.map((user: any) => {
            return {
              key: user.nickname,
              role: user.role[0],
              name: user.name,
              username: user.nickname,
              datelog: user.dateLog,
            };
          })
        );
        setTotalElements(data.totalElements);
      })
      .catch(() => {
        setData([]);
      });
  }, []);
  const changeContent = (page: any) => {
    getUsers(page)
      .then(({ data }: any) => {
        setData(
          data.content.map((user: any) => {
            return {
              key: user.nickname,
              role: user.role[0],
              name: user.name,
              username: user.nickname,
              datelog: user.dateLog,
            };
          })
        );
        setTotalElements(data.totalElements);
      })
      .catch(() => {
        setData([]);
      });
  };
  return (
    <div className="uk-padding-small ">
      <DataTable
        data={data}
        columns={columns}
        responsive
        dense={false}
        disabled={disabled}
        pagination
        paginationServer
        paginationTotalRows={totalElements}
        paginationComponentOptions={options}
        customStyles={customStyles(theme)}
        className="uk-table"
        onChangePage={changeContent}
      />
    </div>
  );
};

export default Users;
