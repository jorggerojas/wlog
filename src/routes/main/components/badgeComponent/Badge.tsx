import React /*useState*/ from "react";

interface BadgeProps {
  data: Array<Object>;
}
const Badge = ({ data }: BadgeProps) => {
  //   const [active, setActive] = useState("false");
  return (
    <ul className="uk-subnav uk-subnav-pill uk-margin-small-left">
      {data.length
        ? data.map((badge: Object) => {
            const pill = JSON.parse(JSON.stringify(badge));
            return (
              <li key={pill.title} className={pill.isActive ? "uk-active" : ""}>
                <a href="#s">{pill.title}</a>
              </li>
            );
          })
        : null}
    </ul>
  );
};

export default Badge;
