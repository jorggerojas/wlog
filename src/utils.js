import React from "react";
import { render, waitFor } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router, Route } from "react-router-dom";
import "@testing-library/jest-dom";

const renderWithRouter = (ui, { route = "/", ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  const finishLoading = () =>
    waitFor(() =>
      expect(utils.queryByText("Loading, please wait...")).toBeNull()
    );
  return {
    ...utils,
    finishLoading,
    history,
  };
};

export {
  Simulate,
  wait,
  render,
  cleanup,
  renderIntoDocument,
  fireEvent,
} from "@testing-library/react";
export { renderWithRouter };

export function renderWithProviders(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
  apolloMocks
) {
  console.log("I am route inside renderWithProviders Wrapper:", route);
  return {
    ...render(
      <Router history={history}>
        <Route path={route}>{ui}</Route>
      </Router>
    ),
    history,
  };
}
