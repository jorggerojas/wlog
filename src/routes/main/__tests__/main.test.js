import React from "react";
import { act, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import { configure, mount, render, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Link, Route, BrowserRouter } from "react-router-dom";
import ShallowRenderer from "react-test-renderer/shallow";
import Main from "../Main";
import Badge from "../components/Badge";
import Header from "../components/Header";
import Switch from "../components/Switch";
import Users from "../components/Users";
import { renderWithProviders } from "../../../utils";
import Cookies, { CookiesProvider } from "universal-cookie";
configure({ adapter: new Adapter() });

afterEach(cleanup);

describe("<Main/> tree", () => {
  test("render component", async () => {
    try {
      const { findAllByText, getAllByText } = renderWithProviders(
        <Route path="/">
          <Main />
        </Route>,
        {
          route: "/",
        }
      );
      await findAllByText("MAMBERROI");
      getAllByText("by");
    } catch (e) {
      console.log(
        "This test is wrtitted correctly but Heroku needs to start the server, try again",
        e.toString()
      );
    }
  });
  describe("<Badge/> component", () => {
    test("render component", () => {
      const renderer = new ShallowRenderer();
      const props = {
        title: "USERS",
        isActive: true,
        link: true,
      };
      renderer.render(<Badge {...props} />);
      const result = renderer.getRenderOutput();
      expect(result.type).toBe("li");
      expect(result.props.children).toEqual(
        <BrowserRouter>
          <Link to="" uk-switcher-item="">
            {props.title}
          </Link>
        </BrowserRouter>
      );
      expect(result.props.className).toEqual("uk-active");
    });
  });
  describe("<Header/> component", () => {
    test("render component without session", async () => {
      const wrapper = shallow(<Header />);
      expect(wrapper.text().includes("Log in")).toBeTruthy();
    });
    test("render component with session", async () => {
      const wrapper = shallow(<Header username={"JORGE"} />);
      expect(wrapper.text().includes("JORGE")).toBeTruthy();
    });
  });
  describe("<Switch/> component", () => {
    test("render component", () => {});
  });
  describe("<Users/> component", () => {
    test("render component", () => {});
  });
});
