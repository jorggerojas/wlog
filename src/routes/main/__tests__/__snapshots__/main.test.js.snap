// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> component render component 1`] = `
<div>
  <div
    class="transparent"
  >
    <div
      class="uk-grid uk-child-width-1-1@s uk-child-width-1-2@m uk-child-width-1-3@l uk-child-width-1-4@xl uk-padding-large"
      uk-grid="masonry: true"
      uk-height-match="target: .card-target"
    >
      <div>
        <div
          class="sc-citwmv giMxTf uk-card-large  uk-card uk-card-default uk-card-hover"
        >
          <div
            class="uk-card-header uk-margin-remove-bottom"
          >
            <h3
              class="sc-iBPRYJ diHnSC uk-card-title uk-margin-remove-bottom"
            >
              <a
                class="uk-link-heading uk-link-reset"
                href="/user/JorggeRojas/post/1"
              >
                FAKE POST
              </a>
            </h3>
            <div
              class="sc-bYEvPH eWQrNj uk-card-badge uk-label uk-text-uppercase uk-visible@s"
            >
              Fake
            </div>
            <div
              class="uk-hidden@s uk-margin-small uk-margin-remove-top"
            >
              <span
                class="sc-gKsewC jbSaSR uk-badge"
              >
                FAKE
              </span>
            </div>
            <p
              class="uk-text-meta uk-text-italic uk-margin-remove-top"
            >
              <time>
                December 16th of 2020
              </time>
               by
               
              <a
                class="uk-button-text"
                href="/user/JorggeRojas"
              >
                JORGGEROJAS
              </a>
            </p>
          </div>
          <div
            class="uk-padding uk-text-break"
          >
            <p
              class="uk-text-italic  uk-visible@s"
            >
              My fake summary
            </p>
            <a
              class="sc-hKgILt evLhsB  uk-button uk-button-text"
              href="/user/JorggeRojas/post/1"
            >
              Read more...
            </a>
          </div>
        </div>
      </div>
      <div>
        <div
          class="sc-citwmv giMxTf uk-card-large  uk-card uk-card-default uk-card-hover"
        >
          <div
            class="uk-card-header uk-margin-remove-bottom"
          >
            <h3
              class="sc-iBPRYJ diHnSC uk-card-title uk-margin-remove-bottom"
            >
              <a
                class="uk-link-heading uk-link-reset"
                href="/user/FakeUser1/post/2"
              >
                MY SECOND POST
              </a>
            </h3>
            <div
              class="sc-bYEvPH eWQrNj uk-card-badge uk-label uk-text-uppercase uk-visible@s"
            >
              Summary
            </div>
            <div
              class="uk-hidden@s uk-margin-small uk-margin-remove-top"
            >
              <span
                class="sc-gKsewC jbSaSR uk-badge"
              >
                SUMMARY
              </span>
            </div>
            <p
              class="uk-text-meta uk-text-italic uk-margin-remove-top"
            >
              <time>
                December 16th of 2020
              </time>
               by
               
              <a
                class="uk-button-text"
                href="/user/FakeUser1"
              >
                FAKEUSER1
              </a>
            </p>
          </div>
          <div
            class="uk-padding uk-text-break"
          >
            <p
              class="uk-text-italic  uk-visible@s"
            >
              This is a really short summary
            </p>
            <a
              class="sc-hKgILt evLhsB  uk-button uk-button-text"
              href="/user/FakeUser1/post/2"
            >
              Read more...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="paginate uk-padding"
    >
      <ul
        class="uk-pagination uk-flex-center"
      >
        <li
          class="uk-pagination-previous uk-disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            role="button"
            tabindex="0"
          >
            Previous
          </a>
        </li>
        <li
          class="uk-pagination-next"
        >
          <a
            aria-disabled="false"
            aria-label="Next page"
            role="button"
            tabindex="0"
          >
            Next
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
